### PR TITLE
Implement product search caching with advanced filtering

### PR DESCRIPTION
**Pull request by mogii**
- Add caching mechanism for product search in bookings controller
- Implement search filtering for cached products in memory
- Support cache invalidation and force refresh
- Add support for doNotCallPluginForProducts configuration
- Enhance test coverage for product search caching

### CODE REVIEW

File: controllers/bookings.js [hunk 1]
```diff
@@ -112,15 +150,68 @@ const $bookingsProductSearch = plugins => async ({
...
+      key: cacheKey,
+    });
+  }
+  const cacheValue = await cache.get({
+    pluginName: appKey,
+    key: cacheKey,
+  });
+  if (cacheValue && cacheValue.products) {
+    const searchResults = $searchProductList(cacheValue.products, searchInput, optionId);
+    console.log(`${appKey}/${userId}/${hint}: found cache and returning cached products: ${searchResults.length}`);
```
Comment (xalakox) [inline controllers/bookings.js:173]: we should have like a debug level log for these, no need o change it; but need to check the logging library we are using (if we are using one)

File: controllers/__tests__/bookings.js [hunk 2]
```diff
@@ -105,21 +117,103 @@ describe('user: booking search', () => {
     // expect(retVal.bookings.length > 0).toBeTruthy();
   });
 
-  it('should be able to search a for a booking product', async () => {
```
Comment (xalakox) [inline controllers/__tests__/bookings.js:108]: we need the default test to also work. clearing the data before / after is relevant but I think is already handled.
Also the force-refresh should be test existing cache data tha differest to see if it calls it, not with empty data.

**Review by xalakox — CHANGES_REQUESTED** [review]
new testing scenarios and keeping the current one would be nice.

File: controllers/__tests__/bookings.js (hunk 2 above)
Comment (mogii) [reply controllers/__tests__/bookings.js:108]: added a test to test no cache and no forceRefresh, which is the default state for all non special tourplan users
